File: Forth_Interpreter/Plus.cpp
#include "Plus.hpp"

#include <memory>

bool Plus::Run () {
    if (GeneralDataForExecutors.Stack->empty ()) {
        throw Exception_EmptyStack();
    }

    int a = GeneralDataForExecutors.Stack->top ();
    GeneralDataForExecutors.Stack->pop ();

    if (GeneralDataForExecutors.Stack->empty ()) {
        throw Exception_EmptyStack();
    }

    int b = GeneralDataForExecutors.Stack->top ();
    GeneralDataForExecutors.Stack->pop ();

    GeneralDataForExecutors.Stack->push(b + a);

    return false;
}

File: Forth_Interpreter_On_Java/Plus/Plus.java
package Plus;

import java.util.EmptyStackException;
import ForthCommands.ForthCommands;

public class Plus extends ForthCommands{

    @Override
    public boolean Run() {
        try {
            int a = GeneralDataForExecutors.stack.peek ();
            GeneralDataForExecutors.stack.pop ();
    
            int b = GeneralDataForExecutors.stack.peek ();
            GeneralDataForExecutors.stack.pop ();
        
            GeneralDataForExecutors.stack.push(b + a);
        } catch (EmptyStackException e) {
            System.err.print(e.getMessage());
        }
    
        return false;
    }
}
